crates/router/src/core/customers.rs

```diff
@@ -319,73 +327,137 @@ pub async fn update_customer(
-    .switch()?;
-
-    let key = key_store.key.get_inner().peek();
-
-    if let Some(addr) = &update_customer.address {
-        let customer_address: api_models::payments::AddressDetails = addr.clone();
-        let update_address = async {
-            Ok(storage::AddressUpdate::Update {
-                city: customer_address.city,
-                country: customer_address.country,
-                line1: customer_address
-                    .line1
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                line2: customer_address
-                    .line2
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                line3: customer_address
-                    .line3
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                zip: customer_address
-                    .zip
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                state: customer_address
-                    .state
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                first_name: customer_address
-                    .first_name
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                last_name: customer_address
-                    .last_name
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                phone_number: update_customer
-                    .phone
-                    .clone()
-                    .async_lift(|inner| types::encrypt_optional(inner, key))
-                    .await?,
-                country_code: update_customer.phone_country_code.clone(),
-            })
-        }
-        .await
-        .switch()
-        .attach_printable("Failed while encrypting Address while Update")?;
-        db.update_address_by_merchant_id_customer_id(
+    let customer = db
+        .find_customer_by_customer_id_merchant_id(
             &update_customer.customer_id,
             &merchant_account.merchant_id,
-            update_address,
             &key_store,
         )
         .await
-        .switch()
-        .attach_printable(format!(
-            "Failed while updating address: merchant_id: {}, customer_id: {}",
-            merchant_account.merchant_id, update_customer.customer_id
-        ))?;
+        .switch()?;
+
+    let key = key_store.key.get_inner().peek();
+
+    let address_id = if let Some(addr) = &update_customer.address {
+        match customer.address_id {
+            Some(address_id) => {
+                let customer_address: api_models::payments::AddressDetails = addr.clone();
+                let update_address = async {
+                    Ok(storage::AddressUpdate::Update {
```

Comment:
Maybe move this implementation to the domain types module?

Reply:
moved this logic to utils